components/function-controller/internal/controllers/serverless/system_state.go

```diff
@@ -140,9 +140,16 @@ func (s *systemState) buildGitJob(gitOptions git.Options, cfg cfg) batchv1.Job {
 	imageName := s.buildImageAddress(cfg.docker.PushAddress)
 
 	args := append(cfg.fn.Build.ExecutorArgs, fmt.Sprintf("%s=%s", destinationArg, imageName), fmt.Sprintf("--context=dir://%s", workspaceMountPath))
-	if s.instance.Spec.RuntimeImageOverride != "" {
-		args = append(args, fmt.Sprintf("--build-arg=base_image=%s", s.instance.Spec.RuntimeImageOverride))
+	if s.instance.Spec.RuntimeImageOverride != nil {
```

Comment:
We should check `RuntimeImageOverride` if it's empty, because for nil and for empty, we shouldn't add arg to list.